tests/integration_tests/core_tests.py

```diff
@@ -1548,6 +1548,30 @@ def test_dashboard_injected_exceptions(self, mock_db_connection_mutator):
         data = self.get_resp(url)
         self.assertIn("Error message", data)
 
+    @mock.patch("superset.sql_lab.cancel_query")
+    def test_stop_query_no_cancel_query(self, mock_sql_lab_cancel_query):
+        """
+        Handles stop query when the DB engine spec does not
+        have a cancel query method.
+        """
+        form_data = {"client_id": "foo"}
+        query_mock = mock.Mock()
+        query_mock.sql = "SELECT *"
+        query_mock.database = 1
+        query_mock.schema = "superset"
+        query_mock.client_id = "foo"
+        query_mock.status = QueryStatus.RUNNING
+        self.login(username="admin")
+
+        with mock.patch("superset.views.core.db") as mock_superset_db:
```

Comment:
Why not also patch the method with a decorator, like you did for `cancel_query`?

Reply:
initially it was because it was giving me a 404 error and I saw this way of doing it repeated in various other integration tests. However, I played around with it, and if I `@mock.patch(superset.views.core.db.session` then I can use it as a decorator in the same way that I use cancel_query. 

So i changed it to that, because ti does make it a lot more consistent and readable.